src/uct/ib/dc/dc_mlx5.c

```diff
@@ -567,12 +570,30 @@ uct_dc_mlx5_init_rx(uct_rc_iface_t *rc_iface,
 
          return uct_rc_mlx5_init_rx_tm(&iface->super, rc_config, &srq_attr,
                                        sizeof(struct ibv_rvh) +
-                                       sizeof(struct ibv_ravh), 0);
+                                       sizeof(struct ibv_ravh));
      }
 #endif
 
+    status = uct_rc_iface_init_rx(rc_iface, rc_config,
+                                  &iface->super.rx.srq.verbs.srq);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    iface->super.rx.srq.type = UCT_IB_MLX5_OBJ_TYPE_VERBS;
```

Comment:
can't you do these 2 assignments before `init_rx`? It would be shorter
```
iface->super.rx.srq.type = UCT_IB_MLX5_OBJ_TYPE_VERBS;
iface->super.super.progress = uct_dc_mlx5_iface_progress;
return uct_rc_iface_init_rx(rc_iface, rc_config, &iface->super.rx.srq.verbs.srq);
```